### ISSUE TITLE
[Bug Report] 多次挂载子组件

### ISSUE THREAD

**Issue by songshuzhong** [ElemeFE/element#21232]
### Element UI version
2.15.5

### OS/Browsers version
chrome 92.0.4515.131 (x86_64)

### Vue version
2.6.14

### Reproduction Link
https://codepen.io/pen/tour/welcome/start

### Steps to reproduce
当el-table-column有fixed属性时，column包裹的子组件会mounted2次并且不会destroy。(当有selection属性时，也会出现这个问题)，例如：
<el-table-column fixed="right">
  <el-alert />
</el-table-column />
alert会mounted 多次。

### What is Expected?
子组件mounted1次，或者mounted后还能执行destroy钩子函数

### What is actually happening?
子组件mounted多次，并且不会执行destroy钩子函数

**Comment by liuning0406** [comment]
I figured out why,be due to "render-after-expand" by default "true",so meet appeared this problem ,let " :render-after-expand='false'",This should resolve the problem.

**Comment by liuning0406** [comment]
However, I found another problem, that is, sometimes the "check-change" event cannot be triggered when the parent node selects and cancles all when clicking the selection box. I don't know why.  Here's the code: 
tree:    <el-tree
        :data="currentShowArr"
        show-checkbox
        node-key="Id"
        accordion
        :render-after-expand="false"
        :default-checked-keys="selectArr"
        @check-change="checkchange"
        ref="tree"
        :props="defaultProps"
      >
      </el-tree>
methods:  // 更改状态后改变数组
    checkchange(a, b, c) {
      console.log(a, b, c, "改变选中", this.changeSelectArr);
      let Idx = this.changeSelectArr.findIndex(item => {
        return item.DepartId == a.Id;
      });
      console.log(Idx, "Idx", this.changeSelectArr);
      debugger;
      if (Idx == -1) {
        this.changeSelectArr.push({
          DepartId: a.Id,
          Name: a.DepartmentName,
          Selected: b
        });
      } else {
        console.log(this.changeSelectArr[Idx], "this.changeSelectArr[Idx]");
        this.changeSelectArr.splice(Idx, 1);
      }
      console.log(this.changeSelectArr, "更改结果");
    },

**Comment by songshuzhong** [comment]
> I figured out why,be due to "render-after-expand" by default "true",so meet appeared this problem ,let " :render-after-expand='false'",This should resolve the problem.

I mean，it is not el-tree but el-table-column。Dose the el-tree have the same problem?

**Comment by liuning0406** [comment]
yes,it is "el-tree".I tried to get the changed array when I checked it, and delete the original changed elements when I unchecked it, but I can get the level 1 directory when I checked it, but the level 1 directory cannot trigger the "check-change" event when I unchecked it, so what do I need to do to make the node at the top level trigger the "check-change" event?

**Comment by liuning0406** [comment]
I'm sorry. I commented in the wrong place

**Comment by liuning0406** [comment]
我很抱歉，我评论错地方了。我提交的bug编号是#21230。我已经将内容及问题移动到那个地方了。
I'm sorry. I commented in the wrong place.  The bug number I submitted is #21230.  I have moved the content and the question to that place.
